.github/workflows/react_tests.yml

```diff
@@ -0,0 +1,50 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: React Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+defaults:
+  run:
+    working-directory: katello
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        path: katello
+    - name: Use Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.6
+    # We could update the postinstall action for foreman to look for an environment variable for plugin webpack dirs
+    # before kicking off the ruby script to find them, setting that would eliminate the ruby dep and running `npm i` in Katello.
+    - name: Install bundler (needed to find Foreman plugins)
+      run: gem install bundler
+    - name: Checkout Foreman repo
+      uses: actions/checkout@v2
+      with:
+        repository: theforeman/foreman
+        path: foreman # checkout Foreman and Katello as siblings, same as dev env
+        ref: develop
+    - name: Install Foreman dependencies
+      run:  npm i
+      working-directory: ./foreman
+    - name: Install Katello dependencies
+      run:  npm i
+    - name: Run Katello tests
+      run: npm test
+    - name: Run Katello linting
```

Comment:
I still think it would be _super_ valuable if this were a separate job, so you could see at a glance if the failure is in linting or tests.

Reply:
It would be nice, but we'll have to think about how to separate out the common actions like the checkout structure and npm installing, probably in another action. 

Seeing as they are pretty quick, maybe we can hold off on that for now?

Reply:
This may be the only time this code will be touched in years.  I think it would be pretty easy to separate out.. you could just rename the existing `build` job to `tests` and add another one (in the same file) for `linting`.  For that you probably could just install Katello only, `npm i`, and then do `npx eslint`.  And the testing and linting would run concurrently!  
If there are any complications at all then I'm fine with the code as-is, but I still think it's worth a shot.

Reply:
thanks for clarifying, I thought you meant to separate out into different actions which would (afaict) require publishing an action to share the common steps.

Eslint does need foreman checked out and installed for a couple reasons: to use the eslint config from foreman, which exists in `vendor-dev` and to check packages for the `    'import/no-extraneous-dependencies'` check. Both of these could maybe be improved in the future but that is the best I can do with the current setup.

I kept the build steps the same and set them up to run the jobs concurrently, currently they aren't working but I'll poke around!